packages/web-components/src/components/pricing-table/pricing-table.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { customElement, property } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+import DDSStructuredList from '../structured-list/structured-list';
+import styles from './pricing-table.scss';
+import DDSPricingTableHeaderCell from './pricing-table-header-cell';
+import DDSPricingTableHighlightLabel from './pricing-table-highlight-label';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+@customElement(`${ddsPrefix}-pricing-table`)
+class DDSPricingTable extends StableSelectorMixin(DDSStructuredList) {
+  @property({ reflect: true, attribute: 'highlight-column' })
+  highlightColumn?: number;
+
+  @property({ reflect: true, attribute: 'highlight-label' })
+  highlightLabel?: string;
+
+  @property()
+  highlightClass = 'highlighted';
+
+  protected _renderHighlightLabel(): DDSPricingTableHighlightLabel {
+    const { highlightLabel } = this;
+    const element = this.ownerDocument.createElement(
+      `${ddsPrefix}-pricing-table-highlight-label`
+    ) as DDSPricingTableHighlightLabel;
+    element.innerText = highlightLabel || '';
+    return element;
+  }
+
+  protected _unhighlightCells(cells: NodeListOf<Element>): void {
+    const { highlightClass } = this;
+    cells.forEach(cell => {
+      cell.classList.remove(highlightClass);
+      cell.querySelector(`${ddsPrefix}-pricing-table-highlight-label`)?.remove();
+      this.style.marginTop = '';
+    });
+  }
+
+  protected _highlightCells(cells: NodeListOf<Element>): void {
+    const { highlightLabel, highlightClass } = this;
+    cells.forEach(cell => cell.classList.add(highlightClass));
+    if (highlightLabel) {
+      const firstCell = cells[0];
+      if (firstCell instanceof DDSPricingTableHeaderCell) {
+        firstCell.prepend(this._renderHighlightLabel());
+      }
+      this._setMarginTop();
+    }
+  }
+
+  protected _setMarginTop(): void {
+    const wrapper = this.shadowRoot?.getElementById(DDSStructuredList.wrapperId) || this;
+    (async () => {
+      return this.querySelector(`${ddsPrefix}-pricing-table-highlight-label`);
+    })()
+      .then(value => {
+        wrapper.style.marginTop = `${value?.getBoundingClientRect().height}px`;
+      })
+      .catch(() => {
+        wrapper.style.marginTop = '';
+      });
```

Comment:
I don't think any part of this is asynchronous - should this just be if/else logic?